### PR TITLE
Rules: git remote add instead of set-url if remote does not exist

### PR DESCRIPTION
**Pull request by PLNech**
screen shot 2016-03-04 at 13 49 37

### CODE REVIEW

**Comment by mcarton** [comment]
Nice, but can we have a test file for that?

**Comment by PLNech** [comment]
@mcarton: I fixed the README test and created a test for this rule.
I wasn't sure which test style I should follow, please tell me if this was not the best way to do so!

**Comment by scorphus** [comment]
Good job, @PLNech :+1: 

Do you mind squashing your three commits into a single one then force-pushing to same branch (your local clone's master, in this case) in order to have a single changeset regarding a single feature?

Thanks!

**Comment by scorphus** [comment]
Also, name the test file after the rule, IOW:
```bash
$ mv test_git_remote_rename_add.py test_git_remote_seturl_add.py
```

File: thefuck/rules/git_remote_seturl_add.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from thefuck.utils import replace_argument
+from thefuck.specific.git import git_support
+
+@git_support
```
Comment (scorphus) [inline thefuck/rules/git_remote_seturl_add.py]: We adopt the style of two lines before a function declaration. Please fix this.

File: thefuck/rules/git_remote_seturl_add.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+def match(command):
+    return('set-url' in command.script and 'fatal: No such remote' in command.stderr)
+
+def get_new_command(command):
```
Comment (scorphus) [inline thefuck/rules/git_remote_seturl_add.py]: Same here :+1: You might already know but it's always worth mentioning Python's [Code Style](http://docs.python-guide.org/en/latest/writing/style/).

**Comment by PLNech** [comment]
@scorphus: thanks for your comments! Good catch for the test rule name, thanks for the style advice - a squashed commit correcting these points is on its way :)

**Comment by scorphus** [comment]
Awesome! :+1:

**Comment by PLNech** [comment]
Done :)

**Comment by scorphus** [comment]
Superb! :+1: 

Oh! Just in time, @PLNech: I forgot to mention your test lacks a `not_match` test case ([exempli gratia](https://github.com/nvbn/thefuck/blob/5389d0c1068ea5081e0a0cdaf14b45a2c6e0463b/tests/rules/test_git_two_dashes.py#L21)).

**Comment by PLNech** [comment]
@scorphus: Thanks, I updated accordingly and resquashed!

**Comment by scorphus** [comment]
Great. Thank you!

**Comment by nvbn** [comment]
Thanks!

**Pull request merged**